modules/console/pages/config/security/authentication.adoc

```diff
@@ -1,61 +1,367 @@
+
+[mermaid]
+....
+flowchart TD
+    A((User authenticates with<br/>OIDC or SASL credentials)) --> B[Redpanda Console]
+
+    B --> C["Re-use OIDC token or basic credentials<br/>(if configured)"]
+    B --> D[Get credentials<br/>from Console config]
+
+    C --> E[Kafka API]
+    C --> F[Admin API]
+    C --> G[Schema API]
+    E & F & G --> H((Redpanda))
+
+    D --> J[Kafka Connect API]
+    D --> L[Git]
+....
+
+== Supported authentication methods
+
+Redpanda Console supports two login authentication methods:
+
+* **OIDC:**
+  Uses external identity providers, such as Google, to facilitate single sign-on (SSO).
+* **Basic authentication:**
+  Provides a traditional username and password login.
+
+== How authentication works in Redpanda Console
+
+When a user logs in successfully through OIDC or basic authentication, Redpanda Console creates a secure session by generating a JWT. This JWT is signed with a secret defined by `jwtSigningKey` and, if configured, stored as a secure cookie. The token encapsulates the user's credentials, which is either an ID token from an OIDC provider or the username/password provided by basic authentication, and serves as the basis for authenticating API requests.
+
+[mermaid]
+....
+flowchart TD
+    A[User login attempt] --> B{Select auth method}
+
+    %% OIDC flow
+    B -- OIDC --> C[Redirect to identity provider]
+    C --> D[User authenticates & receives ID token]
+    D --> E[Generate JWT with ID token]
+
+    %% Basic flow
+    B -- Basic --> F[Submit username/password]
+    F --> G[Validate credentials]
+    G --> H[Generate JWT with credentials]
+
+    %% Unified session
+    E & H --> I[Store JWT as secure cookie]
+    I --> J[User makes API call]
+
+    %% API selection
+    J --> K{Which API?}
+    K -- "Kafka API" --> L[Check Kafka impersonation setting]
+    K -- "Schema registry API" --> M[Check schema registry impersonation setting]
+    K -- "Admin API" --> N[Check admin API impersonation setting]
+
+    %% Kafka API branch
+    subgraph KafkaFlow [Kafka API flow]
+      L --> O{User impersonation enabled?}
+      O -- Yes --> P["Forward JWT credentials\n(SASL-OAUTHBEARER or SASL-SCRAM)"]
+      O -- No --> Q[Use static credentials\n+ enforce console RBAC]
+    end
+
+    %% Schema registry branch
+    subgraph SchemaRegistryFlow [Schema registry API flow]
+      M --> R{User impersonation enabled?}
+      R -- Yes --> S[Forward JWT credentials\nheader-based authentication]
+      R -- No --> T[Use static credentials\n+ enforce console RBAC]
+    end
+
+    %% Admin API branch
+    subgraph AdminAPIFlow [Admin API flow]
+      N --> U{User impersonation enabled?}
+      U -- Yes --> V[Forward JWT credentials\nheader-based authentication]
+      U -- No --> W[Use static credentials\n+ enforce console RBAC]
+    end
+....
+
+=== OIDC authentication
```

Comment:
Great, concise and accurate description. I was wondering whether

1. Redpanda itself also must configure OIDC auth for the different APIs
2. The OIDC configs must match, there's no point in using different OIDC providers

Both are kind of obvious, but maybe people may forget about configuring OIDC in Redpanda as well.

Reply:
Nevermind this is already done in the prerequisites further below